v2/pkg/parsers/parser.go

```diff
@@ -59,12 +59,12 @@ func LoadWorkflow(templatePath string) (bool, error) {
 	return false, nil
 }
 
-func isTemplateInfoMetadataMatch(tagFilter *filter.TagFilter, templateInfo *model.Info, extraTags []string) (bool, error) {
+func isTemplateInfoMetadataMatch(tagFilter *filter.TagFilter, templateInfo *model.Info, extraTags []string, templateType string) (bool, error) {
```

Comment:
I think nothing explicitly stops someone to declare multiple protocols in one template. If it's the case then it should be in plural (otherwise explicit validation is required that would prevent this from happening) and instead of `type` it should rather be `protocol` and instead of string, an "enum" type

Reply:
Converted it to an enum as per your suggestion